### PR TITLE
[REF] website_variants_extra: place product price, product qty, and "add to cart "btn in  3rd column. It also add the reviews number, Icons(quote, Ask question,Print) and the behavior to share products . The product name change acording to the selected variants.  HU#1006 ISSUE#571

### PR DESCRIPTION
**Pull request by gmlp**
**BUILD:**  https://github.com/Vauxoo/yoytec/pull/675
@oscarolar, these are the changes applied in website_variants_extra module.
Best regards

### CODE REVIEW

**Comment by oscarolar** [comment]
Reference this PR to the plan please.
and create a dummy build with yoytec env.

**Comment by nhomar** [comment]
@gmlp 
1. Your commit meessage says "website_yoytec:" which is a module that not belong to this PR.

**Comment by gmlp** [comment]
screenshot from 2016-02-15 13-08-37
@oscarolar

**Comment by oscarolar** [comment]
asfd

**Comment by oscarolar** [comment]
I cant see the manufacturer model and the SKU, did you miss it?

**Comment by oscarolar** [comment]
What was wrong with the icons the customer provided?
sad

File: website_variants_extra/views/templates.xml [hunk 1]
```diff
@@ -45,12 +46,125 @@
...
+                                    </a>
+                                </span>
+                                <input type="text" class="js_quantity form-control" data-min="1" name="add_qty" value="1"/>
+                                <span class="input-group-addon">
+                                    <a t-attf-href="#" class="mb8 float_left js_add_cart_json">
+                                        <i class="fa fa-plus"></i>
+                                    </a>
+                                </span>
+                            </div>
+                            <a id="add_to_cart" class="btn btn-warning btn-lg mt16 js_check_product a-submit" href="#">Add to Cart</a>
```
Comment (oscarolar) [inline website_variants_extra/views/templates.xml]: Can you remove the class `btn-lg` to make the button fit the size asked by the customer.

**Comment by gmlp** [comment]
@oscarolar , the icons asked don't appears in https://fortawesome.github.io/Font-Awesome/icons/

**Comment by oscarolar** [comment]
A lot of space between text on availability, set the `line-height` style to a smaller value.
s

**Comment by oscarolar** [comment]
I still see a pill on the 'Extra price' section, as the customer asked it must be a rectangle
nopill

**Comment by oscarolar** [comment]
I see black some text that should be gray
nopill

File: website_variants_extra/views/templates.xml [hunk 2]
```diff
@@ -45,12 +46,125 @@
...
+                                    <tr>
+                                        <td><span class="fa fa-money"/></td>
+                                        <td><span class="sale-action">Quote</span></td>
+                                    </tr>
+                                    <tr>
+                                        <td><span class="fa fa-heart-o"/></td>
+                                        <td><span class="sale-action">Add To Wishlist</span></td>
+                                    </tr>
+                                    <tr>
+                                        <td class="text-center"><span class="fa fa-question"/></td>
```
Comment (oscarolar) [inline website_variants_extra/views/templates.xml]: use `fa fa-question-circle`

**Comment by oscarolar** [comment]
I cant see the share icons, I have mentioned before that there is a simple implementation on how to sahre the  product, since there are only two lines of code to add in order to get this working I suggest that you only copy the lines of [this module](https://github.com/OCA/e-commerce/blob/8.0/website_product_share/views/website_product.xml#L8-L9) instead of adding a dependency and installing an extra module.
shareicons

**Comment by oscarolar** [comment]
I cant see the breadcrumb on your PR, what happened?
no

File: website_rate_product/__openerp__.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "website": "http://www.vauxoo.com/",
     "license": "AGPL-3",
     "depends": [
-        "website_sale"
```
Comment (oscarolar) [inline website_rate_product/__openerp__.py:10]: why did you remove this and added the other one?

**Comment by gmlp** [comment]
@oscarolar sku and manufacturer belong to another repo

File: website_variants_extra/static/src/less/website.less [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+#add_to_cart{
+    background-color:#f39c12;
+}
+#product_with_variants{
+    color: @link-color;
+}
+.js_quantity.form-control {
+    background-color: white;
+
+}
```
Comment (oscarolar) [inline website_variants_extra/static/src/less/website.less:73]: please put all these classes namespaced, do not leave them just as simple CSS rules, this mean you have to nest them under a parent selector.

File: website_variants_extra/views/templates.xml [hunk 5]
```diff
@@ -45,12 +46,125 @@
...
 		  	<attribute name="class">col-sm-4 col-md-4 col-lg-3 col-lg-offset-1 variants_column</attribute>
 		  </xpath>
-<!-- 		  <xpath expr="//t[@t-placeholder='select']" position="replace">
+ 		  <xpath expr="//t[@t-placeholder='select']" position="replace">
 		    <input type="hidden" t-if="len(product.product_variant_ids) == 1" name="product_id" t-att-value="product.product_variant_ids[0].id"/>
               <t t-call="website_variants_extra.variants">
               </t>
-		  </xpath> TODO: THIS CODE IS COMMENTED BECAUSE IT CRASHES PHANTOM CRAWLER, 
-		  				 IT HAS A PURPOSE DISCUSS WITH NHOMAR--> 
+		  </xpath> TODO: THIS CODE IS COMMENTED BECAUSE IT CRASHES PHANTOM CRAWLER,
```
Comment (oscarolar) [inline website_variants_extra/views/templates.xml]: This is no longer happening, please delete the code

File: website_rate_product/views/star_rate.xml [hunk 6]
```diff
@@ -9,8 +9,8 @@
             </xpath>
         </template>
 
-        <template id="rating_product" inherit_id="website_sale.product" active="False" customize_show="True" name="Show Product Rating">
-            <xpath expr="//h1[@itemprop='name']" position="after">
+        <template id="rating_product" inherit_id="website_variants_extra.product_variants_description_3columns" active="False" customize_show="True" name="Show Product Rating">
```
Comment (gmlp) [inline website_rate_product/views/star_rate.xml]: why did you remove this and added the other one?
@oscarolar ,Because, i replace all the right div and change the target to website_variants_extra, therefore this module depends of website_variants_extra

Comment (oscarolar) [inline website_rate_product/views/star_rate.xml]: I'm talking about dependencies you removed the dependency `website_sale`and added `website_variants_extra`, this has nothing to do with the divs or whatever, besides, if I want to install the module `website_rate_product`only, I can't, you have to inherit `website_rate_product` from `website_variants_extra` and then do the correct inheritance. https://github.com/Vauxoo/addons-vauxoo/pull/708/files#diff-2aa691e2f5886e33884e188617cd0711L10

**Comment by gmlp** [comment]
screenshot from 2016-02-16 17-07-30

@oscarolar , dude change done

**Comment by nhomar** [comment]
@gmlp Please answer the first @oscarolar request:

Create a Dommy PR on customer repository and reference it here, Please.

**Comment by gmlp** [comment]
@nhomar,
**workplan:** https://github.com/vauxoo/yoytec/pull/640
**dummy:**  https://github.com/Vauxoo/yoytec/pull/675

**Comment by oscarolar** [comment]
@gmlp regarding the icon theme please use the ones provided by the customer https://github.com/Vauxoo/yoytec/issues/571

**Comment by gmlp** [comment]
@oscarolar , changes done.
screenshot from 2016-02-18 11-47-58

**Comment by gmlp** [comment]
screenshot from 2016-02-18 18-34-06
@oscarolar , improvements done.

File: website_variants_extra/static/src/js/website_sale.js [hunk 7]
```diff
@@ -1,48 +1,38 @@
...
+                else{
+                    variants_str = $(this).next().html();
+                }
+            });
+            var $product_name = $('#product_with_variants').html();
+            var index = $product_name.indexOf(",");
+            if(index > -1){
+                $product_name = $product_name.substring(0,index);
+            }
+            $('#product_with_variants').html($product_name+", "+variants_str);
```
Comment (oscarolar) [inline website_variants_extra/static/src/js/website_sale.js]: this JS must be tested with PhantomJS, please make a test that change selected attributes and verfy that it build the name correctly.

File: website_variants_extra/static/src/js/website_sale.js [hunk 8]
```diff
@@ -1,48 +1,38 @@
...
-    $('ul.js_add_cart_variants', oe_website_sale).each(function () {
         $('input.js_variant_change, select.js_variant_change', this).first().trigger('click');
     });
-});
-});
+    openerp.website.if_dom_contains('#no-reviews', function(){
+        var comments_qty = $('#comments-list').children().length;
+        if (comments_qty > 0){
+            var concept = comments_qty >= 2 ? "reviews" : "review";
+            $('#no-reviews').text(comments_qty+" "+concept);
```
Comment (oscarolar) [inline website_variants_extra/static/src/js/website_sale.js]: what if there are comments in the backend? this algorithm will work?, take in account that if you post a comment in the backend it will be rendered on the frontend even   (unpublished), I don't see any validation for unpublished comments.

Comment (gmlp) [inline website_variants_extra/static/src/js/website_sale.js]: @oscarolar, it's only happens with  super users accounts that select which comments be posted.

File: website_product_availability/models/product.py [hunk 9]
```diff
@@ -151,11 +151,12 @@ def _get_planned_dates(self):
     @api.multi
     def _product_availability_warehouse(self, warehouse_id):
         # search avalailability for stokable products
-        if self.type not in('product'):
+        if self.type not in 'product':
             return True
         domain_quant = []
         product = self
         domain_quant += [('product_id', 'in', [product.id])]
+        # pylint: disable=unused-variable
```
Comment (oscarolar) [inline website_product_availability/models/product.py]: This was made because the method returns 3 variables but we need only one.

**Comment by oscarolar** [comment]
:+1:

File: website_product_availability/models/product.py [hunk 10]
```diff
@@ -151,11 +151,12 @@ def _get_planned_dates(self):
     @api.multi
     def _product_availability_warehouse(self, warehouse_id):
         # search avalailability for stokable products
-        if self.type not in('product'):
+        if self.type not in 'product':
```
Comment (nhomar) [inline website_product_availability/models/product.py]: el in es para comparar iterables, haber usado una tupla no está mal.... o usas == o usas in con un iterable. ambas opciones son válidas, tu cambio está incorrecto.

File: website_product_availability/models/product.py (hunk 9 above)
Comment (nhomar) [inline website_product_availability/models/product.py]: Si no está siendo usada borrala

File: website_rate_product/models/mail_message.py [hunk 11]
```diff
@@ -27,7 +27,7 @@
 
 class MailMessage(osv.Model):
     _inherit = 'mail.message'
-    _columns = {
+    _columns = {  # pylint: disable=attribute-deprecated
```
Comment (nhomar) [inline website_rate_product/models/mail_message.py]: usa nueva api.

los pylint disabled es cuando no se puede resolver algo, evitenlos a lo máximo por favor, esto es estúpido que sea hecho de eta manera.

**Comment by gmlp** [comment]
@oscarolar and @nhomar ,changes done

**Comment by nhomar** [comment]
@gmlp Can you please before merge:

1.- Rebase this with base addons-vauxo.
2.- What I understand in the first 2 commit one of them fixe the other one. (or at least you mention something which I did not understand was "Impossible".
3.- Please update de description of the PR in order to si the Before - After with this feature, to many messages in the middle make to me impossible to understand the real conclusion, make a good resume of the job achieved with this PR.
4.- You mentiondes an HU in the title of the PR but I can not see a mention the #CA which you are covering.

Thanks in advance.

**Comment by gmlp** [comment]
@oscarolar and @nhomar ,
Dummy: https://github.com/Vauxoo/yoytec/pull/675
Berofe:
screenshot from 2016-03-01 11-53-11
After:
screenshot from 2016-03-01 11-42-05

File: website_variants_extra/static/src/less/website.less [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+        padding-right: 30px;
+        padding-top: 5px;
+        padding-bottom: 5px;
+    }
+
+    .website-sale-actions .custom-icon {
+        width: 30px;
+    }
+    .website-sale-actions .icon-quote {
+        width: 20px;
```
Comment (nhomar) [inline website_variants_extra/static/src/less/website.less]: Usar less:

``` less
.website-sale-actions {
  .custom-icon {
    width: 30px;
  }  
  .icon-quote {
    width: 30px;
  }
}
```

File: website_variants_extra/static/src/js/test_product_name_with_variants.js [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+                            },
+                waitFor:   '#product_with_variants:contains(100w)',
+                waitNot:   '#product_with_variants:contains(200w)',
+                element:    '.js_add_cart_variants span:contains(Blue)',
+            },
+        ],
+    });
+
+}());
+
```
Comment (nhomar) [inline website_variants_extra/static/src/js/test_product_name_with_variants.js:88]: No logro ver donde pruebas el "Quantity of comments."

File: website_variants_extra/static/src/less/website.less [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+        font-size: 18px;
+        .stars{
+            color: #a94442;
+            margin-right: 5px;
+        }
+    }
+    .product_header{
+        color: @link-color;
+    }
+    hr{
```
Comment (nhomar) [inline website_variants_extra/static/src/less/website.less]: `hr {` Sin el espacio esto dará problemas tarde o temprano.

File: website_variants_extra/static/src/less/website.less [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+.actions_column{
+    .small-text {
+        .text-green{
+            color: @link-color;
+        }
+    }
+    .js_quantity.form-control {
+        background-color: white;
+    }
+    #add_to_cart{
```
Comment (nhomar) [inline website_variants_extra/static/src/less/website.less]: intenta usar clases no #

File: website_variants_extra/views/templates.xml [hunk 16]
```diff
@@ -3,135 +3,135 @@
...
+                    </div>
+                    </form>
 
-		      </li>
-		    </t>
-		  </ul>
-		</div>
+                </div>
+                </div>
+		  </xpath>
```
Comment (nhomar) [inline website_variants_extra/views/templates.xml]: identaciones limpias pliz.

**Comment by nhomar** [comment]
@gmlp mucho mejor... solo comentarios menores.

File: website_variants_extra/views/templates.xml (hunk 1 above)
Comment (oscarolar) [inline website_variants_extra/views/templates.xml]: Not valid anymore.
